model/src/main/kotlin/AnalyzerResult.kt

```diff
@@ -86,8 +90,21 @@ class AnalyzerResultBuilder(
     }
 
     fun addResult(projectAnalyzerResult: ProjectAnalyzerResult) = this.apply {
-        projects.add(projectAnalyzerResult.project)
-        packages.addAll(projectAnalyzerResult.packages)
-        errors[projectAnalyzerResult.project.id] = projectAnalyzerResult.errors
+        val duplicateProject = projects.find { it.id == projectAnalyzerResult.project.id }
+        if (duplicateProject != null) {
+            val error = "Found multiple projects with the same id '${duplicateProject.id}', not adding the " +
+                    "duplicate to the analyzer results. The definition files containing the duplicate projects are " +
+                    "'${duplicateProject.definitionFilePath}' and " +
+                    "'${projectAnalyzerResult.project.definitionFilePath}'."
+
+            log.error { error }
+
+            val projectErrors = errors.getOrDefault(duplicateProject.id, listOf())
```

Comment:
Is it a good idea to merge the errors here? If there really are different errors in the duplicate, it's confusing so see them in the original, as you would then look into the original for debugging and not be able to reproduce the error.

Reply:
Actually only the error about the duplicate `id` is added to avoid exactly this confusion. `projectAnalyzerResult.errors` is ignored in this case.

Reply:
Ah, thanks for pointing that out.